packages/common/src/Utility.ts

```diff
@@ -739,7 +739,10 @@ export function textSize(_text: string | string[], fontName: string = "Verdana",
         g_fontSizeContext.font = `${bold ? "bold " : ""}${fontSize}px ${fontName}`;
         g_fontSizeContextCache[hash] = retVal = {
             width: Math.max(...text.map(t => g_fontSizeContext.measureText("" + t).width)),
-            height: fontSize * text.length
+            height: Math.max(...text.map(t => {
+                const textMeasurement = g_fontSizeContext.measureText("" + t);
+                return textMeasurement.fontBoundingBoxDescent + textMeasurement.fontBoundingBoxAscent;
```

Comment:
Based upon these docs https://developer.mozilla.org/en-US/docs/Web/API/TextMetrics#examples

Reply:
1.  The result still needs to be multiplied by `text.length`
2. (optional) The calculation for width and height could be moved into a single `text.reduce` function.

Reply:
I couldn't think of a way to make `text.reduce` work, because the type of the accumulator must that of the individual items in the array, rather than whatever you want, eg an object containing the dimensions.